src/components/connect.js

```diff
@@ -17,9 +17,36 @@ function getDisplayName(WrappedComponent) {
 // Helps track hot reloading.
 let nextVersion = 0
 
-export default function connect(mapPropsToRequestsToProps, options = {}) {
+function connectFactory(options = {}) {
+  function connectImpl(map) {
+    return connect(map, options)
+  }
+
+  connectImpl.withFetch = function (fetch) {
+    invariant(
+      typeof fetch === 'function',
+      '`withFetch` must be passed a function. Instead received a %s.',
+      typeof fetch
+    )
+
+    return connectFactory(assign({}, options, { fetch: fetch }))
+  }
+
+  connectImpl.withRef = function (withRef = true) {
+    return connectFactory(assign({}, options, { withRef: withRef }))
+  }
+
+  return connectImpl
+}
+
+export default connectFactory()
+
+function connect(mapPropsToRequestsToProps, options) {
   const finalMapPropsToRequestsToProps = mapPropsToRequestsToProps || defaultMapPropsToRequestsToProps
-  const { withRef = false } = options
+  const {
+    fetch = window.fetch,
```

Comment:
yes i agree with dropping the `window` part (one difference is it will throw a ReferenceError if it isn't defined, ie. if the browser doesn't have it and the user forgot to include a polyfill, but this is probably a plus actually)